api/src/domain/tax-filings/taxFilingsInterfaceFactory.test.ts

```diff
@@ -140,83 +140,81 @@ describe("TaxFilingsInterfaceFactory", () => {
         });
       });
 
-      it("returns lookup response when lookup returns API_ERROR", async () => {
+      it("returns lookup response when onboarding returns SUCCESS", async () => {
+        const filingData = generateTaxFiling({});
+        taxFilingClient.onboarding.mockResolvedValue({
+          state: "SUCCESS",
+          filings: [filingData],
+        } as TaxFilingOnboardingResponse);
         taxFilingClient.lookup.mockResolvedValue({
-          state: "API_ERROR",
-          filings: [],
+          state: "SUCCESS",
+          filings: [filingData],
```

Comment:
I think that because these two calls have the same mocked response value, it doesn't feel like the test is actually testing that it "returns the lookup response"

Like, if Lookup returns ObjectA, and Onboarding also returns the same ObjectA - if the expectation is that we get back ObjectA, it's not actually validating if that's the thing that was returned from Lookup or from Onboarding. If your code was written such that it just passed through the response from the onboarding success without ever calling lookup, I think this test would still pass. It's not validating that the thing that's getting returned is specifically the return from the lookup call.